### PR TITLE
Webpack Extensions update

### PR DESCRIPTION
**Pull request by smartguest**
This PR adds in webpacking for the external extensions that previously weren't (with the exception of import as that currently requires a module that is incompatible with webpacking). 

The changes include:
 1. adding/editing the appropriate .vscodeignore and .gitignore files, webpack config files (also cleaning up entries that are no longer relevant as well) 
 
 2. bumping the localization extension to 4.1.2. to work with webpack (it does not affect functionality of the extension).  
 
 3. Additionally for Server-report, there was an old renderTemplateHtml function that has been removed as handlebar requires a complicated workaround to webpack, and it was not used at all inside ADS nor in the extension itself. (The extension has been verified to work fine without it)
 
 Extensions that have been modified to use this process include:
- [x] 'admin-tool-ext-win',
- [x] 'agent',
- [x] 'cms',
- [x] 'dacpac',
- [x] 'machine-learning',
- [x] 'profiler',
- [x] 'schema-compare',
- [x] 'server-report',
- [x] 'sql-assessment',
- [x] 'sql-database-projects'
- [x] 'sql-migration'
- [x] 'query-history'
- [x] 'resource-deployment' - Minor fix in vscodeIgnore

If this requires a bump in the extension version number, please let me know (Server-report is already bumped to a higher number).

### CODE REVIEW

File: extensions/admin-tool-ext-win/extension.webpack.config.js [hunk 1]
```diff
@@ -2,6 +2,16 @@
  *  Copyright (c) Microsoft Corporation. All rights reserved.
  *  Licensed under the Source EULA. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
+
```
[the remark this answers is not in the document]
Comment (smartguest) [reply extensions/admin-tool-ext-win/extension.webpack.config.js:5]: Yes I did, but the only string contained in it was for the renderTemplateHTML which no longer exists.

**Review by Charles-Gagnon — COMMENTED** [review]
Let's make sure we test each of these extensions to verify that they still work as expected. Can you add a list of the changed extensions in this PR and then do a private build to generate the VSIXs for people to test with?

**Comment by smartguest** [comment]
@Charles-Gagnon here are the extensions that were changed, I added it to my initial PR message:

	'admin-tool-ext-win',
	'agent',
	'cms',
	'dacpac',
	'machine-learning',
	'profiler',
	'schema-compare',
	'server-report',
	'sql-assessment',
	'sql-database-projects'
	
I already have an ad-hoc build with the extensions compiled, should I post the link here or give it to the authors of the extensions to test?

**Comment by Charles-Gagnon** [comment]
Thanks! I added checkboxes so we can check them off as people test. 

Probably easiest to just link here and tag extension owners (let me know if you're unsure who an owner is)

**Comment by smartguest** [comment]
Hello all,

There have been some changes made to the extension files allowing them to be webpacked, can you please test them to make sure they still work as expected? Thanks.

The extension files are published here as part of the drop: https://mssqltools.visualstudio.com/CrossPlatBuildScripts/_build/results?buildId=108947&view=artifacts&pathAsName=false&type=publishedArtifacts

@abist 
@anthonydresser 
@aasimkhan30 
@kisantia 
@kburtram 
@llali 
@alanrenmsft 
@udeeshagautam 
@vchvlad 
@SakshiS-harma

**Review by kisantia — APPROVED** [review]

**Review by abist — APPROVED** [review]
Tested and looks fine for agent and cms

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**